doc/man7/flux-broker-attributes.rst

```diff
@@ -34,22 +34,34 @@ version
 
 rundir [Updates: C]
    A temporary directory where the broker's UNIX domain sockets are located.
-   In addition, if ``statedir`` is not set, this directory is used by the
-   content backing store (if applicable).  By default, each broker rank creates
-   a unique ``rundir`` in ``$TMPDIR`` and removes it on exit.  If ``rundir`` is
+   By default, each broker rank creates
+   a unique temporary directory and removes it on exit.  If ``rundir`` is
    set on the command line, beware exceeding the UNIX domain socket path limit
    described in :linux:man7:`unix`, as low as 92 bytes on some systems.  To
    support the :man1:`flux-start` ``--test-size`` option where multiple brokers
    share a ``rundir``, if ``rundir`` is set to a pre-existing directory, the
    directory is not removed by the broker on exit.  In most cases this
    attribute should not be set by users.
 
+rundir-cleanup
+   The broker sets this attribute to ``1`` if the run directory is a
+   temporary directory that will be removed when the broker exits.
+   It sets it to ``0`` if the directory was supplied on the command line
+   and will persist after the broker exits.
+
 statedir [Updates: C]
-   A directory in which persistent state is stored by the Flux broker.  For
-   example, content backing store data is stored here to facilitate restarts.
-   If unset, this data goes to ``rundir`` where it is cleaned up on instance
-   shutdown.  If set, this directory must exist and be owned by the instance
-   owner.  Default: unset.
+   A directory in which persistent state is stored by the Flux leader broker.
+   For example, content backing store data is stored here to facilitate
+   restarts.  If unset, a unique temporary directory is created that
+   is cleaned up on instance shutdown.
+   If set, this directory must exist and be owned by the instance owner.
+   The state directory is only used on the leader (rank 0) broker.
```

Comment:
Is this correct? It seems like the broker will create the last component of `statedir`
```console
$ flux batch -N1 --output=test.out --broker-opts=-Sstatedir=/var/tmp/grondo/state-{{id}} --wrap flux getattr statedir
f23zjhCzB
$ cat test.out
/var/tmp/grondo/state-f23zjhCzB
$ flux batch -N1 --output=test.out --broker-opts=-Sstatedir=/var/tmp/grondo/flux/state-{{id}} --wrap flux getattr statedir
f4RHpTdif
$ cat test.out
lt-flux-broker: statedir error creating /var/tmp/grondo/flux/state-f4RHpTdif: No such file or directory
```

Reply:
Yep the docs are not correct - good catch!

The broker also removes the directory if it creates it.  Now I'm thinking that's wrong, it should only remove it if the directory was not specified and it came from `mkdtemp()`.

Reply:
> The broker also removes the directory if it creates it. Now I'm thinking that's wrong, it should only remove it if the directory was not specified and it came from mkdtemp().

Ok, and we'd rely on other cleanup mechanisms in the case of e.g. `-Sstatedir=/l/ssd/{{jobid}}`?

Reply:
I actually wasn't thinking about this for that use case because {{jobid}} cannot be guaranteed to be unique.  I was thinking instead to add a follow-up PR that allowed another tmpdir like directory (like `/l/ssd`) to be specified in another option and then let the broker would use` mkdtemp()` in that instead of in TMPDIR or /var/tmp.

My concern about removing a directory that the user actually named is that they might expect it to persist with its contents.

Reply:
That makes sense. Note that in the use case specified in the original issue {{jobid}} would be guaranteed to be unique because the user knows they aren't running more subinstances within their batch job (or those do not need a manually specified `statedir`)

However, having a way to easily ask for a `statedir` to be created in a different filesystem than the default is a much more user friendly solution!